src/Runner/Runner.php

```diff
@@ -83,8 +89,36 @@ public function __construct(
         $this->isDryRun = $isDryRun;
 
         $this->cacheManager = new FileCacheManager(
-            $config->usingCache(),
-            $config->getCacheFile(),
+            $this->cacheHandler($config),
+            $this->signature($config),
+            $isDryRun
+        );
+    }
+
+    /**
+     * @param ConfigInterface $config
+     *
+     * @return HandlerInterface
+     */
+    private function cacheHandler(ConfigInterface $config)
```

Comment:
naming convension, it's not a getter to use shorthand method name, it creates a new instance. please rename the method, like `createFoo`